internal/tools/orchestrator/endpoints/runtime.go

```diff
@@ -727,13 +729,33 @@ func (e *Endpoints) GetAppWorkspaceReleases(ctx context.Context, r *http.Request
 }
 
 func (e *Endpoints) KillPod(ctx context.Context, r *http.Request, vars map[string]string) (httpserver.Responser, error) {
+	userID, err := user.GetUserID(r)
+	if err != nil {
+		return apierrors.ErrGetAppWorkspaceReleases.NotLogin().ToResp(), nil
+	}
+
 	var req apistructs.RuntimeKillPodRequest
 	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
 		return apierrors.ErrKillPod.InvalidParameter("req body").ToResp(), nil
 	}
-	err := e.runtime.KillPod(req.RuntimeID, req.PodName)
-	if err != nil {
+
+	audit := apistructs.Audit{
```

Comment:
There should be something similar to `auditutil.InitFromHTTPRequest(r *http.Request)` here.

Reply:
> There should be something similar to `auditutil.InitFromHTTPRequest(r *http.Request)` here.

The audit provider internal/pkg/audit is already in place, awaiting @haiqingchq upcoming refactor.

Reply:
TODO list already comment.